plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -214,42 +195,63 @@ sealed class CodeScanSessionConfig(
             if (selectedFile.path.startsWith(projectRoot.path)) {
                 files.addAll(
                     VfsUtil.collectChildrenRecursively(projectRoot).filter {
-                        (it.path.endsWith(sourceExt[0]) || (sourceExt.getOrNull(1) != null && it.path.endsWith(sourceExt[1]))) && it != selectedFile
+                        it != selectedFile && !it.isDirectory && !ignoreFile(it)
                     }
                 )
             }
             return files
         }
     }
 
-    open fun isProjectTruncated() = isProjectTruncated
+    fun isProjectTruncated() = isProjectTruncated
 
-    protected fun getPath(root: String, relativePath: String = ""): Path? = try {
+    fun getPath(root: String, relativePath: String = ""): Path? = try {
         Path.of(root, relativePath).normalize()
     } catch (e: Exception) {
         LOG.debug { "Cannot find file at path $relativePath relative to the root $root" }
         null
     }
 
-    protected fun File.toVirtualFile() = LocalFileSystem.getInstance().findFileByIoFile(this)
+    fun File.toVirtualFile() = LocalFileSystem.getInstance().findFileByIoFile(this)
+
+    private fun convertGitIgnorePatternToRegex(pattern: String): String = pattern
+        .replace(".", "\\.")
+        .replace("*", ".*")
+        .let { if (it.endsWith("/")) "$it?" else it }
+
+    private fun parseGitIgnore(): List<String> {
+        var gitIgnoreFile = File(projectRoot.path, ".gitignore")
+        if (!gitIgnoreFile.exists()) {
+            return emptyList()
+        }
+        return gitIgnoreFile.readLines()
+            .filterNot { it.isBlank() || it.startsWith("#") }
+            .map { it.trim() }
+            .map { convertGitIgnorePatternToRegex(it) }
+    }
```

Comment:
what happened to deduping this

Reply:
hey richard...added this directly into the class. do you want me to change it to seperate class or a util object?

Reply:
Added a new ProjectFileManager class to include all the stuff deduped from weaverbird.